scripts/diff_changes.py

```diff
@@ -0,0 +1,132 @@
+    'upgrade2018_realistic' : '**2018 realistic**',
+    'upgrade2018_realistic_RD' : '**2018 Run-dependent MC**',
+    'upgrade2018_realistic_HEfail' : '**2018 realistic (HEM15/16 failure)**',
+    'upgrade2018cosmics_realistic_deco' : '**2018 cosmics (tracker deco mode)**',
+    'upgrade2018cosmics_realistic_peak' : '**2018 cosmics (tracker peak mode)**',
+    'upgrade2018_realistic_HI' : '**2018 heavy ion**',
+    'upgrade2021_design' : '**2021 design**',
+    'upgrade2021_realistic' : '**2021 realistic**',
+    'upgrade2021cosmics_realistic_deco' : '**2021 cosmics**',
+    'mcRun3_2021_design' : '**2021 design**',
+    'mcRun3_2021cosmics_realistic_deco' : '**2021 cosmics**',
+    'mcRun3_2021_realistic' : '**2021 realistic**',
+    'mcRun3_2021_realistic_HI' : '**2021 heavy ion**',
+    'mcRun3_2023_realistic' : '**2023 realistic**',
+    'mcRun3_2024_realistic' : '**2024 realistic**',
+    'upgrade2023_realistic' : '**2023 realistic**',
+    'mcRun4_realistic' : '**Phase 2 realistic**'}
+
+def global_tags(new_gt):
+    """Get a list of global tags that have
+    changed since release."""
+    file = "Configuration/AlCa/python/autoCond.py"
+    manipulations = " | grep '>' | grep X | cut -d ':' -f 2 | sed 's/ //g' | cut -d ',' -f 1"
+    manipulations += " | sed \"s/'//g\""
+    if new_gt:
+        manipulations = manipulations.replace('>', '<')
+    cmd = "diff " + os.environ["CMSSW_BASE"] + "/src/" + file
+    cmd += " " + os.environ["CMSSW_RELEASE_BASE"] + "/src/" + file
+    output = os.popen(cmd + manipulations).read()
+    # remove trailing newline as well
+    return output[:-1].split('\n')
+
+def checkEnvironment():
+
+    currDir = os.getcwd()
+
+    if not "CMSSW" in currDir:
+        sys.exit('!! WARNING !! this script should be executed from the $CMSSW_BASE/src dirctory, exiting!')
+
+    if "CMSSW_BASE" not in os.environ:
+        sys.exit('!! WARNING !! This script should be executed after invoking the \'cmsenv\' command, exiting!')
+
+    cmsswBase = os.environ["CMSSW_BASE"]
+    if cmsswBase+'/src' != currDir:
+        sys.exit('!! WARNING !! This script should be executed in the same $CMSSW_BASE/src directory used to invoke the \'cmsenv\' command, exiting!')
+
+    if not os.path.isdir(currDir+'/Configuration/AlCa'):
+        sys.exit('!! WARNING !! Please checkout the Configuration/AlCa package before running this script, exiting!')
+
+def main():
+    """Performs the difference between the GTs in your CMSSW release
+    directory (including the modifications) with those in $CMSSW_RELEASE_BASE
+    and prints out a list of URLs showing the set of GT diffs for use in a PR description."""
+    checkEnvironment()
+    new_gts = global_tags(True)
+    old_gts = global_tags(False)
+    diff_links_base = "https://cms-conddb.cern.ch/cmsDbBrowser/diff/Prod/gts/"
+    diff_links = []
+
+    for new_gt in new_gts:
+        gt_base = re.sub('_v[0-9]*', '_', new_gt)
+        if int(gt_base.split('_')[0][0:2]) < 111:
```

Comment:
so if one doesn't change anything in the local installation, the script will end in error.

Reply:
Indeed, I just managed to reproduce the error. I will fix it. Thanks for spotting it!